### PR TITLE
fix(build): hashes in prod builds now changes when ID change.

### LINKED ISSUE #2868: Wrong hashing for `scripts` production bundle
> ## Please provide us with the following information:
### OS?

> Windows 7, 8 or 10. Linux (which distribution). Mac OSX (Yosemite? El Capitan?)

Linux x64 (Ubuntu 16.04)
### Versions.

> Please run `ng --version`. If there's nothing outputted, please run in a Terminal: `node --version` and paste the result here:

angular-cli: 1.0.0-beta.17
node: 6.5.0
### Repro steps.

> Was this an app that wasn't created using the CLI? What change did you do on your code? etc.

Using the CLI for dev/prod for more than a month. Updating to latest version continously.
### The log given by the failure.

> Normally this include a stack trace and some more information.

No errors related. (Only webpack production warnings)
### Mention any other details that might be useful.

The hashed name generated for `scripts.**.bundle.js` generated for different versions of the same app (like adding a service) is the same while the bundle content has differences (seems like chunk numbers generated by webpack).

Example flow to replicate:
create an app
add a script to `angular-cli.json`
generate a production bundle <-- generates a `scripts` bundle with a hash X.
add some TS module to the app (like a new component)
generate a production bundle <-- generates a `scripts` bundle with the same hash X but the content is different (in particular, webpack-related chunk numbers).

This affects production caching since the bundle cannot be invalidated by name since the hash works 'fake' (same hash, different content, enough to make the app fail to load).

> ---
> 
> Thanks! We'll be in touch soon.

### LINKED ISSUE #3311: hash code not changing in inline.js file during build
> Please provide us with the following information:
> ---------------------------------------------------------------

Any OS

1.0.0-beta.21

I have an app with lazy loaded modules. I made a change in one of modules and run build.prod. I noticed that bundled file has changed (also it's hash code in filename has changed), as well as content of inline.bundle.js file has changed updating module bundle file hash code. But hash code for inline.bundle.js file is the same as previous so it's loading cached file inline.bundle.js instead of updated file.

no log here

### Mention any other details that might be useful.

Related to #2899 
> ---------------------------------------------------------------
> Thanks! We'll be in touch soon.

### PR DESCRIPTION
**Pull request by hansl**
Fixes #2868.
Fixes #3311.

### CODE REVIEW

File: tests/e2e/tests/build/chunk-hash.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  return Promise.resolve()
+    .then(() => ng('generate', 'module', 'lazy', '--routing'))
+    .then(() => addImportToModule('src/app/app.module.ts', oneLine`
+      RouterModule.forRoot([{ path: "lazy", loadChildren: "./lazy/lazy.module#LazyModule" }])
+    `, '@angular/router'))
+    .then(() => ng('build', '--prod'))
+    .then(() => {
+      fs.readdirSync('./dist')
+        .forEach(name => {
+          if (!name.match(/(main|inline|style|\d+)\.[a-z0-9]+\.bundle\.js/)) {
```
Comment (filipesilva) [inline tests/e2e/tests/build/chunk-hash.ts]: The styles bundle should be `dist/styles.[hash].bundle.css` instead of `dist/style.[hash].bundle.js`.

**Comment by filipesilva** [comment]
lgtm aside from the styles check.

File: tests/e2e/tests/build/chunk-hash.ts (hunk 1 above)
Comment (hansl) [inline tests/e2e/tests/build/chunk-hash.ts]: done.

**Comment by elvirdolic** [comment]
Amazing!!!

**Pull request merged**